### DISCUSSION TITLE
[Feature Request] <Add ABI Information to zkSync Explorer>

### DISCUSSION THREAD

**Discussion by samui68400** [zkSync-Community-Hub/zksync-developers#474] [Feature requests]
### Network

zkSync Era

### Provide a brief description of the feature you would like to see implemented.

I suggest adding ABI information to the zkSync Explorer to enhance usability and accessibility for users and developers interacting with smart contracts on the zkSync netw

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

This feature request primarily benefits developers and users who rely on the zkSync Explorer to interact with smart contracts. Having access to ABI information directly in the Explorer streamlines contract interaction and improves the overall developer experience.

### Additional Details

I have explored an alternative solution by using encoded data through the CLI, but this method is not user-friendly and adds complexity to the coding process. Integrating ABI information directly into the zkSync Explorer would significantly improve the user experience and streamline contract interaction.

**Comment by bxpana** [comment]
Thanks for the feature request! Will add that to our list of things and work on implementation. If you know more people who want this you can have them upvote it so we can see what the community wants us to prioritize

**Comment by samui68400** [comment]
Thanks for the answer!

**Comment by samui68400** [comment]
I just ask to the community on discord what they think about it, let's see!

**Comment by Cryptostart1** [comment]
that would be useful indeed

**Comment by dutterbutter** [comment]
+1 for this! and thank @samui68400 for adding a feature ticket to the repo itself: https://github.com/matter-labs/block-explorer/issues/242

**Reply by samui68400** [reply]
It's a great pleasure to learn, and if I can contribute, it's even more fun!

**Reply by samui68400** [reply]
If I can help with development, that's fine with me. I think I'd know what to do. I just need a little help if I have any questions on how to do it.

**Reply by dutterbutter** [reply]
Repo is open-sourced so certainly feel free to contribute! Its always appreciated :)
